project/src/test/java/com/google/sps/EventDetailTest.java

```diff
@@ -103,6 +107,34 @@ public void verifyGetAllEvents() throws IOException {
         CommonUtils.convertToJson(expectedEvents).trim(), stringWriter.toString().trim());
   }
 
+  @Test
+  public void verifyGetFilteredEvents() throws IOException {
+    Set<Event> expectedEvents = new HashSet(Arrays.asList(EVENT1, EVENT2));
+    Mockito.when(request.getParameter("labelParams")).thenReturn( "Work-Chess");
+
+    new FilteredEventServlet().doGet(request, response);
+
+    try {
+      JSONAssert.assertEquals(CommonUtils.convertToJson(expectedEvents).trim(),stringWriter.toString().trim(), /*ordered=*/ false);
+    } catch (JSONException e) {
+      System.out.println("JSON conversion failed.");
+    }
+  }
+
+  @Test
+  public void verifyGetOneFilteredEvent() throws IOException {
+    Set<Event> expectedEvents = new HashSet(Arrays.asList(EVENT1));
+    Mockito.when(request.getParameter("labelParams")).thenReturn( "Work");
+
+    new FilteredEventServlet().doGet(request, response);
```

Comment:
You can initialize the Servlet in setup to make it a bit cleaner.

Reply:
ditto here. Leave something and then resolve it.

Reply:
The servlet setup annotation is @BeforeClass not @Before, so I think it doesnt apply